nuxeo-platform-video-core/src/main/java/org/nuxeo/ecm/platform/video/tools/VideoSlicer.java

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * The {@link VideoTool} for slicing video blobs.
+ *
+ * @since 8.4
+ */
+public class VideoSlicer extends VideoTool {
+
+    public final static String NAME = "slicerTool";
+
+    public final static String START_AT_PARAM = "startAt";
+
+    public final static String DURATION_PARAM = "duration";
+
+    public final static String ENCODE_PARAM = "encode";
+
+    protected final static String SEGMENTS_PATH = "segmentsPath";
+
+    protected static final String COMMAND_SLICER_DEFAULT = "videoSlicer";
+
+    protected static final String COMMAND_SLICER_BY_COPY = "videoSlicerByCopy";
+
+    protected static final String COMMAND_SLICER_SEGMENTS = "videoSlicerSegments";
+
+    protected static final String COMMAND_SLICER_START_AT = "videoSlicerStartAt";
+
+    public VideoSlicer() {
+        super(NAME, COMMAND_SLICER_DEFAULT);
+    }
+
+    @Override
+    public Map<String, String> setupParameters(BlobHolder blobHolder, Map<String, Object> parameters) {
+        Map<String, String> cmdParameters = super.setupParameters(blobHolder, parameters);
+        Blob video = blobHolder.getBlob();
+        String startAt = (String) parameters.get(START_AT_PARAM);
+        String duration = (String) parameters.get(DURATION_PARAM);
+        boolean encode = (boolean) parameters.get(ENCODE_PARAM);
+
+        boolean useStart = !StringUtils.isEmpty(startAt);
+        boolean useDuration = !StringUtils.isEmpty(duration);
+        if (useStart) {
+            cmdParameters.put(START_AT_PARAM, startAt);
+
+            String finalFilename = setupFilename(video.getFilename(), startAt, duration);
+            File outputFile;
+            try {
+                outputFile = Framework.createTempFile(FilenameUtils.removeExtension(finalFilename),
+                        "sliced." + FilenameUtils.getExtension(finalFilename));
+            } catch (IOException e) {
+                throw new NuxeoException("VideoSlicer could not set up temporary output file.");
+            }
+            Framework.trackFile(outputFile, this);
+            cmdParameters.put(OUTPUT_FILE_PATH_PARAM, outputFile.getAbsolutePath());
+
+            // check if the slicer should use the duration
+            if (useDuration) {
+                cmdParameters.put(DURATION_PARAM, duration);
+                commandLineName = encode? COMMAND_SLICER_DEFAULT : COMMAND_SLICER_BY_COPY;
+            } else {
+                commandLineName = COMMAND_SLICER_START_AT;
+            }
+        } else if (useDuration) {
+            Path segmentsDirectory;
+            try {
+                segmentsDirectory = Framework.createTempDirectory("Segments");
+            } catch (IOException e) {
+                throw new NuxeoException("VideoSlicer could not create temporary directory for video segments");
+            }
+            File segments = segmentsDirectory.toFile();
+            File dir = new File(segments.getAbsolutePath(), addSuffixToFileName(video.getFilename(), "-%03d"));
```

Comment:
They are being tracked in the `buildResult` method, because in the `setupParameters` we don't know which files (and how many) are going to be created.